bin/oneoff/nces_data/import_school_stats_by_years_2023_2024_ccd

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env ruby
+
+require_relative '../../../dashboard/config/environment'
+
+CDO.log = Logger.new($stdout)
+
+SURVEY_YEAR = '2023-2024'.freeze
+SURVEY_YEAR_HEADER = '2023-24'.freeze
+LAST_SURVEY_YEAR = '2022-2023'.freeze
+
+# Override for the dry run variable using an commandline argument
+DRY_RUN = !(ARGV.find {|arg| arg.casecmp('-dryrun')}).nil?
+
+VIRTUAL_SCHOOL_MAP = {
+  'FULLVIRTUAL' => 'Yes',
+  'NOTVIRTUAL' => 'No',
+  'SUPPVIRTUAL' => 'No',
+  'FACEVIRTUAL' => 'Yes',
+  '–' => nil,
+  '†' => nil
+}.freeze
+
+COMMUNITY_TYPE_MAP = {
+  '11' => 'city_large',
+  '12' => 'city_midsize',
+  '13' => 'city_small',
+  '21' => 'suburban_large',
+  '22' => 'suburban_midsize',
+  '23' => 'suburban_small',
+  '31' => 'town_fringe',
+  '32' => 'town_distant',
+  '33' => 'town_remote',
+  '41' => 'rural_fringe',
+  '42' => 'rural_distant',
+  '43' => 'rural_remote'
+}.freeze
+
+GRADES_MAP = {
+  'Prekindergarten' => 'PK',
+  'Kindergarten' => 'KG',
+  '1st Grade' => '01',
+  '2nd Grade' => '02',
+  '3rd Grade' => '03',
+  '4th Grade' => '04',
+  '5th Grade' => '05',
+  '6th Grade' => '06',
+  '7th Grade' => '07',
+  '8th Grade' => '08',
+  '9th Grade' => '09',
+  '10th Grade' => '10',
+  '11th Grade' => '11',
+  '12th Grade' => '12',
+  '13th Grade' => '13',
+  'Adult Education' => 'AE',
+  'Ungraded' => 'UG',
+  '–' => 'M',
+  '†' => 'N'
+}
+
+TITLE_I_STATUS_MAP = {
+  'Title I targeted assistance eligible school--No program' => '1',
+  'Title I targeted assistance school' => '2',
+  'Title I schoolwide eligible--Title I targeted assistance program' => '3',
+  'Title I schoolwide eligible school--No program' => '4',
+  'Title I schoolwide school' => '5',
+  'Not a Title I school' => '6',
+  'Missing' => nil
+}
```

Comment:
New this year since the `title_i_status` from NCES was formatted as one of the strings above rather than the status value.